### PR TITLE
Add pre-fill for the command handler tests

### LINKED ISSUE #14: Pre-fill command handler tests [requires #10]
Currently autogenerated command handler tests look something like this:

```
class <ActionName><EntityName>HandlerTest extends TestCase
{
    use ProphecyTrait;
    use <EntityName>TestTrait;
    use TransactionManagerTestTrait;
    use DomainEventTestTrait;

    private <ActionName><EntityName>Handler $fixture;

    protected function setUp () : void
    {
        $this->init<EntityName>TestTrait();
        $this->initTransactionManagerTestTrait();
        $this->initDomainEventTestTrait();

        // TODO inject dependencies
        $this->fixture = new <ActionName><EntityName>Handler();
        $this->fixture->setTransactionManager($this->transactionManager->reveal());
        $this->fixture->setEventRegistry($this->eventRegistry->reveal());
    }

    // TODO implement test cases for <ActionName><EntityName>HandlerTest 
}
```

99% of command handlers have a workflow that includes loading the entity from the repository, performing some action on it, and then returning the entity. Once #10 is implemented, it would be cool to extend command handler test generation so that the following test method and helper methods are added to the end:
```
    public function test_TESTED_LOGIC_Is_ACTIONED_And<EntityName>IsDequeuedByEventRegistry() : void
    {
        $<entityName>Id = $this->given<EntityName>Id();
        // TODO add additional command params
        $<entityName> = $this->given<EntityName>CanBeFoundById($<entityName>Id);
        $this->then_SOMETHING_ShouldBe_ACTIONED_($<entityName>, /** TODO additional params **/);
        $this->then<EntityName>ShouldBeDequeuedByEventRegistry($<entityName>->reveal());
        $this->when<CommandClassName>IsHandled($<entityName>Id, /** TODO additional command params **/);
    }

    private function then_SOMETHING_ShouldBe_ACTIONED_(ObjectProphecy | <EntityName> $<entityName>, /** TODO additional params **/) : void
    {
        $<entityName>->_DO_ACTION_(/** TODO add params **/)->shouldBeCalled();
    }

    private function when<CommandClassName>IsHandled(<EntityName>Id $<entityName>Id, /** TODO additional command params **/) : void
    {
        $this->fixture->handle(new <CommandClassName>($<entityName>Id, /** TODO additional command params **/));
    }
```
The above code block also requires adding two use statements at the top of the test file for the following classes:
* `Prophecy\Prophecy\ObjectProphecy`
* The command class

Also, in the `setUp` method, replace `$this->fixture = new <ActionName><EntityName>Handler();`  with:
```
$this->fixture = new <ActionName><EntityName>Handler(
    $this-><entityName>Repository->reveal(),
);
```

Finally, add the following method to the entity test trait:
```
    protected function then<EntityName>ShouldBeDequeuedByEventRegistry($<entityName>) : void
    {
        $this->thenEventRegistryShouldDequeueAndRegister($<entityName>);
    }
```
The lack of argument typing in this method is intentional, to allow for various argument wildcarding via Prophecy.

### PR DESCRIPTION
**Pull request by oechsler**

### CODE REVIEW

**Comment by oechsler** [comment]
Do we need translation stuff here?

File: src/Resources/skeleton/ddd/command-handler-test.tpl.php [hunk 1]
```diff
@@ -31,9 +33,31 @@ protected function setUp () : void
...
 
         // TODO inject dependencies
-        $this->fixture = new <?= $extra["command_namespace"]; ?>Handler();
+        $this->fixture = new <?= $extra["command_namespace"]; ?>Handler(
+            $this-><?= $strtocamel($entity); ?>Repository->reveal()
+        );
         $this->commandHandlerPostSetUp();
     }
 
+    public function test_TESTED_LOGIC_Is_ACTIONED_And<?= $entity; ?>IsDequeuedByEventRegistry() : void
```
Comment (msvujnovic-becklyn) [inline src/Resources/skeleton/ddd/command-handler-test.tpl.php]: `_TESTED_LOGIC_Is_ACTIONED_` -> `_TESTED_LOGIC_IS_ACTIONED_`

Translation for 'And' & 'IsDequeuedByEventRegisty' is needed

File: src/Resources/skeleton/ddd/command-handler-test.tpl.php [hunk 2]
```diff
@@ -31,9 +33,31 @@ protected function setUp () : void
...
+    {
+        $<?= $strtocamel($entity); ?>Id = $this->given<?= $entity; ?>Id();
+        // TODO add additional command params
+        $<?= $strtocamel($entity); ?> = $this->given<?= $entity; ?>CanBeFoundById($<?= $strtocamel($entity); ?>Id);
```
Comment (msvujnovic-becklyn) [inline src/Resources/skeleton/ddd/command-handler-test.tpl.php]: Translation for 'given' and 'CanBeFoundById' is needed

File: src/Resources/skeleton/ddd/command-handler-test.tpl.php [hunk 3]
```diff
@@ -31,9 +33,31 @@ protected function setUp () : void
...
+        $this->then_SOMETHING_ShouldBe_ACTIONED_($<?= $strtocamel($entity); ?> /** TODO additional params **/);
```
Comment (msvujnovic-becklyn) [inline src/Resources/skeleton/ddd/command-handler-test.tpl.php]: Translation for 'then' and 'ShouldBe' is needed

File: src/Resources/skeleton/ddd/command-handler-test.tpl.php [hunk 4]
```diff
@@ -31,9 +33,31 @@ protected function setUp () : void
...
+        $this->then<?= $entity; ?>ShouldBeDequeuedByEventRegistry($<?= $strtocamel($entity); ?>->reveal());
```
Comment (msvujnovic-becklyn) [inline src/Resources/skeleton/ddd/command-handler-test.tpl.php]: Translation for 'then' and 'ShouldBeDequeuedByEventRegistry' is needed

File: src/Resources/skeleton/ddd/command-handler-test.tpl.php [hunk 5]
```diff
@@ -31,9 +33,31 @@ protected function setUp () : void
...
+        $this->when<?= $extra["command_namespace"]; ?>IsHandled($<?= $strtocamel($entity); ?>Id /** TODO additional command params **/);
```
Comment (msvujnovic-becklyn) [inline src/Resources/skeleton/ddd/command-handler-test.tpl.php]: Translation for 'when' and 'IsHandled' is needed

File: src/Resources/skeleton/ddd/command-handler-test.tpl.php [hunk 6]
```diff
@@ -31,9 +33,31 @@ protected function setUp () : void
...
+    }
+
+    private function then_SOMETHING_ShouldBe_ACTIONED_(ObjectProphecy | <?= $entity; ?> $<?= $strtocamel($entity); ?> /** TODO additional params **/) : void
```
Comment (msvujnovic-becklyn) [inline src/Resources/skeleton/ddd/command-handler-test.tpl.php]: Translation for 'then' and 'ShouldBeDequeuedByEventRegistry' is needed

File: src/Resources/skeleton/ddd/command-handler-test.tpl.php [hunk 7]
```diff
@@ -31,9 +33,31 @@ protected function setUp () : void
...
+    {
+        $<?= $strtocamel($entity); ?>->_DO_ACTION_(/** TODO add params **/)->shouldBeCalled();
+    }
+
+    private function when<?= $extra["command_namespace"]; ?>IsHandled(<?= $entity; ?>Id $<?= $strtocamel($entity); ?>Id /** TODO additional command params **/) : void
```
Comment (msvujnovic-becklyn) [inline src/Resources/skeleton/ddd/command-handler-test.tpl.php]: Translation for 'when' and 'IsHandled' is needed

File: src/Resources/skeleton/ddd/entity-test-trait.tpl.php [hunk 8]
```diff
@@ -63,4 +63,9 @@ protected function <?= $i18n["test"]["_given"]; ?><?= $entity; ?><?= $i18n["test
     {
         $this-><?= $strtocamel($entity); ?>Repository->findOneById($<?= $strtocamel($entity); ?>Id)->willThrow(new <?= $entity; ?><?= $i18n["not_found"]; ?>Exception());
     }
+
+    protected function then<?= $entity; ?>ShouldBeDequeuedByEventRegistry($<?= $strtocamel($entity); ?>) : void
```
Comment (msvujnovic-becklyn) [inline src/Resources/skeleton/ddd/entity-test-trait.tpl.php]: Translation for 'then' and 'ShouldBeDequeuedByEventRegistry' is needed

**Comment by msvujnovic-becklyn** [comment]
Yep, I noted all the places I could see which need translation. Some of the parts which need translation already have it implemented, others you will need to create.

**Review by msvujnovic-becklyn — CHANGES_REQUESTED** [review]

**Review by glassp — APPROVED** [review]
Should be good after the changes Marko requested

File: src/Resources/i18n/de.yaml [hunk 9]
```diff
@@ -15,16 +15,21 @@ test:
     _when: wenn           # when (gherkin scheme)
     _then: dann         # then (gherkin scheme)
 
+    and: Und
     be_returned: ErwiedertWerden
     by_id: DurchId
     can_be_found_by_id: KannDurchIdGefundenWerden
     cannot_be_found_by_id: KannDurchIdNichtGefundenWerden
+    dequeued_by_event_registry: VonDerEventRegisitrierungAusDerWarteschlangeEntfernt
```
Comment (msvujnovic-becklyn) [inline src/Resources/i18n/de.yaml]: Bwah :D

Machma lieber `VonDerEventRegistryDequeued`, weil die event registry ist da auf Englich in der Codebasis, und wir benutzen auch Englische begriffe wenn wir drüber reden.

**Review by msvujnovic-becklyn — CHANGES_REQUESTED** [review]

File: src/Resources/i18n/de.yaml (hunk 9 above)
Comment (oechsler) [reply src/Resources/i18n/de.yaml]: Ok, kann auch nix dafür, dass Deutsch dreck is 🤣

**Review by msvujnovic-becklyn — APPROVED** [review]

**Pull request merged**